### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by JayKowski**
a form system built with Rails in the back and front-end. ready for review

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following issues need to be fixed before moving on ⬇️** 

* [x] it is possible to edit a user in app **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/edit_user.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

* [x] Routes file should have `resources :users, only: [:new, :create, :edit, :update]` in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

##### `edit` action

* [x] finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

##### `update` action

* [x] finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
* [x] `redirect_to edit_user_path` if user was saved
* [x] `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

#### Views

##### `new`

* [x] commented out `form` uses `users_path` helper for action.

`edit`
* [x] Should have the same commented out forms as the `new` view.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @user = User.new
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      # flash[:notice] = "Article was successfully edited"
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Delete the comment

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  def create
+    # @user = User.new( username: params[:username], email: params[:email], password: params[:password])
+    @user = User.new(user_params)
+    if @user.save
+      # flash[:notice] = "User was successfully created"
+      redirect_to user_path(@user)
+    else
+      render :new
+    end
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Delete the comments

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

:wave: you did an awesome job and thank you for applying the suggested changes.
However, there are some issues that should be fixed before moving on,

1. I got this error when creating a new user
Screenshot from 2020-01-21 11-06-23
Please, read that requirements
Screenshot from 2020-01-21 11-14-07

2. Delete the comment except for the ones in views

Kindly make the changes and submit a new code review request.

Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

You can move on now :+1: 
Good luck with the next.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**